database_cleanup/models/purge_tables.py

```diff
@@ -0,0 +1,110 @@
+        'cleanup.purge.wizard.table', 'Purge Wizard', readonly=True)
+
+    @api.multi
+    def purge(self):
+        """
+        Unlink tables upon manual confirmation.
+        """
+        if self:
+            objs = self
+        else:
+            objs = self.env['cleanup.purge.line.table']\
+                .browse(self._context.get('active_ids'))
+        tables = objs.mapped('name')
+        for line in objs:
+            if line.purged:
+                continue
+
+            # Retrieve constraints on the tables to be dropped
+            # This query is referenced in numerous places
+            # on the Internet but credits probably go to Tom Lane
+            # in this post http://www.postgresql.org/\
+            # message-id/22895.1226088573@sss.pgh.pa.us
+            # Only using the constraint name and the source table,
+            # but I'm leaving the rest in for easier debugging
+            self.env.cr.execute(
+                """
+                SELECT conname, confrelid::regclass, af.attname AS fcol,
+                    conrelid::regclass, a.attname AS col
+                FROM pg_attribute af, pg_attribute a,
+                    (SELECT conname, conrelid, confrelid,conkey[i] AS conkey,
+                         confkey[i] AS confkey
+                     FROM (select conname, conrelid, confrelid, conkey,
+                       confkey, generate_series(1,array_upper(conkey,1)) AS i
+                       FROM pg_constraint WHERE contype = 'f') ss) ss2
+                WHERE af.attnum = confkey AND af.attrelid = confrelid AND
+                a.attnum = conkey AND a.attrelid = conrelid
+                AND confrelid::regclass = '%s'::regclass;
+                """, (IdentifierAdapter(line.name, quote=False),))
+
+            for constraint in self.env.cr.fetchall():
+                if constraint[3] in tables:
+                    self.logger.info(
+                        'Dropping constraint %s on table %s (to be dropped)',
+                        constraint[0], constraint[3])
+                    self.env.cr.execute(
+                        "ALTER TABLE %s DROP CONSTRAINT %s",
+                        (
+                            IdentifierAdapter(constraint[3]),
+                            IdentifierAdapter(constraint[0])
+                        ))
+
+            self.logger.info(
+                'Dropping table %s', line.name)
+            self.env.cr.execute(
+                "DROP TABLE %s", (IdentifierAdapter(line.name),))
+            line.write({'purged': True})
+        return True
+
+
+class CleanupPurgeWizardTable(models.TransientModel):
+    _inherit = 'cleanup.purge.wizard'
+    _name = 'cleanup.purge.wizard.table'
+    _description = 'Purge tables'
+
+    @api.model
+    def find(self):
+        """
+        Search for tables that cannot be instantiated.
+        Ignore views for now.
+        """
+        # Start out with known tables with no model
+        known_tables = ['wkf_witm_trans']
```

Comment:
With the removing of the workflows, this table can and should now be purged as well.